### PR TITLE
TF-1573: Cursor auto move while typing

### PR DESCRIPTION
**Pull request by ManhNTX**
Need merge https://github.com/linagora/enough_html_editor/pull/19 before review
- Device small:

https://user-images.githubusercontent.com/107173849/225802456-3b61dc9b-03f8-4e8d-82d5-3a84a3510c68.mov

- Device big:

https://user-images.githubusercontent.com/107173849/225802483-9fbecfbe-6116-4df9-ad4d-8d07de615681.mov

- landscape mode :

https://user-images.githubusercontent.com/107173849/225812007-47928546-21d4-493f-833d-bbf683a7a84c.mov

- double cursor Android:

https://user-images.githubusercontent.com/107173849/225825188-13fa8e7c-fcfb-47c3-9b7c-4a313e1aa24f.mp4

### CODE REVIEW

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -1500,6 +1506,30 @@ class ComposerController extends BaseController {
...
+      if (coordinates?[1] != null && coordinates?[1] != 0) {
+        final coordinateY = max((coordinates?[1] ?? 0) - defaultPaddingCoordinateYCursorEditor, 0);
+        final realCoordinateY = coordinateY + (headerEditorMobileSize?.height ?? 0);
+        if (scrollController.position.pixels >= realCoordinateY) {
+          scrollController.jumpTo(
+            realCoordinateY.toDouble() - (headerEditorMobileSize?.height ?? 0) / 2,
+          );
+        }
+
+        if ((realCoordinateY) >= max(Get.height - maxKeyBoardHeight - richTextBarHeight, 0) + scrollController.position.pixels) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:1524]: Please separate the handler function to clarify the meaning.

**Review by dab246 — COMMENTED** [review]
Please check also case `landscape` mobile

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -1505,6 +1519,37 @@ class ComposerController extends BaseController {
     }
   }
 
+  void removeFocusAllInputEditorHeader() {
+    subjectEmailInputFocusNode?.unfocus();
+    toAddressFocusNode?.unfocus();
+    ccAddressFocusNode?.unfocus();
+    bccAddressFocusNode?.unfocus();
+  }
+  void _onChangeCursorOnMobile(List<int>? coordinates) {
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:1528]: Add line spacing

File: lib/features/composer/presentation/composer_view.dart [hunk 3]
```diff
@@ -467,29 +469,41 @@ class ComposerView extends GetWidget<ComposerController>
...
+            ? _buildFromEmailAddress(context)
+            : const SizedBox.shrink()),
+        Obx(() => controller.identitySelected.value != null
+            ? const Divider(color: AppColor.colorDividerComposer, height: 1)
+            : const SizedBox.shrink()),
+        _buildEmailAddress(context),
+        const Divider(color: AppColor.colorDividerComposer, height: 1),
+        Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
+            child: _buildSubjectEmail(context)),
```
Comment (nqhhdev) [inline lib/features/composer/presentation/composer_view.dart:499]:
```suggestion
        Obx(() => controller.identitySelected.value != null
          ? _buildFromEmailAddress(context)
          : const SizedBox.shrink()),
        Obx(() => controller.identitySelected.value != null
          ? const Divider(color: AppColor.colorDividerComposer, height: 1)
          : const SizedBox.shrink()),
        _buildEmailAddress(context),
        const Divider(color: AppColor.colorDividerComposer, height: 1),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
          child: _buildSubjectEmail(context)),
```

**Review by dab246 — COMMENTED** [review]
Double cursor error still exists @ManhNTX 


https://user-images.githubusercontent.com/80730648/225910966-5015e2b5-41c7-4c95-a67f-59585f0e3070.mov

**Review by dab246 — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**